### PR TITLE
Bug/77582 able to continue partway through address search

### PR DESCRIPTION
**Pull request by tapiwamukwewa**
#### Checklist

- [ ] Includes tests
- [ ] Update documentation

#### Reviewers should focus on:

added onValidatePostCode prop (as function) so we can pass the valid/invalid state of the postcode to the containing component.

### CODE REVIEW

**Comment by sussexrick** [comment]
@tapiwamukwewa I added the following to the `AddressLookup` component in `addressLookup.mdx`:

```
onValidatePostCode={(isValid) => console.log("isValid changed to " + isValid) }
```

With this in place when I click on "I need to change the address" I get "RangeError: Maximum call stack size exceeded", which sounds like there's probably an infinite loop going on.

I wonder whether setting a `postcodeValid` property, similar to the loading property, rather than calling a function might be a better approach? @vrojastpr may have some useful advice on that.

File: packages/forms/src/elements/address/postcodeLookup.tsx [hunk 1]
```diff
@@ -44,9 +46,20 @@ export const PostcodeLookup: React.FC<PostcodeLookupProps> = ({
...
+		if(typeof result === 'undefined'){
+			setPostcodeValid(true);
+			isValid=true;
+		}else{
+			setPostcodeValid(false);
+			isValid=false;
+		}
+
+		if(onValidatePostCode !== null){
+			validatePostcode(isValid);
```
Comment (allanthompson-tpr) [inline packages/forms/src/elements/address/postcodeLookup.tsx:60]: Should be onValidatePostCode(isValid) I think?

File: packages/forms/src/__tests__/addressLookup.spec.tsx [hunk 2]
```diff
@@ -40,6 +40,7 @@ const defaultProps: AddressProps = {
 	postcodeLabel: 'Postcode',
 	countryLabel: 'Country',
 	changeAddressButton: 'I need to change the address',
+	onValidatePostCode: null,
```
Comment (allanthompson-tpr) [inline packages/forms/src/__tests__/addressLookup.spec.tsx:43]: I wonder if its possible to test the new behaviour (assert that the onValidatePostCode method is called if the postcode is changed?).

File: packages/layout/src/components/cards/common/views/address/addressPage.tsx [hunk 3]
```diff
@@ -79,6 +79,7 @@ const AddressPage: React.FC<AddressPageProps> = ({
 									changeAddressButton={i18n.changeAddressButton}
 									findAddressCancelledButton={i18n.findAddressCancelledButton}
 									onFindAddressCancelled={onCancelChanges}
+									onValidatePostcode ={null}
```
Comment (vrojastpr) [inline packages/layout/src/components/cards/common/views/address/addressPage.tsx:82]: Once you've made `onValidatePostcode` optional, and none of the cards is using this value, we could safely remove this from the props and not need to pass `null`.

**Review by vrojastpr — APPROVED** [review]

**Pull request merged**